core/src/avm1/globals/netconnection.rs

```diff
@@ -1,26 +1,153 @@
 use crate::avm1::function::{Executable, FunctionObject};
+use crate::avm1::globals::shared_object::{deserialize_value, serialize};
 use crate::avm1::object::Object;
-use crate::avm1::property::Attribute;
 use crate::avm1::property_decl::{define_properties_on, Declaration};
-use crate::avm1::{Activation, Error, ScriptObject, TObject, Value};
+use crate::avm1::{
+    Activation, ActivationIdentifier, Error, ExecutionReason, NativeObject, ScriptObject, TObject,
+    Value,
+};
 use crate::avm1_stub;
-use crate::context::GcContext;
+use crate::context::{GcContext, UpdateContext};
+use crate::net_connection::{NetConnectionHandle, NetConnections, ResponderCallback};
+use crate::string::AvmString;
+use flash_lso::packet::Header;
+use flash_lso::types::Value as AMFValue;
+use gc_arena::{Collect, Gc};
+use ruffle_wstr::WStr;
+use std::cell::Cell;
+use std::collections::BTreeMap;
+use std::rc::Rc;
+
+#[derive(Clone, Debug, Collect)]
+#[collect(require_static)]
+struct NetConnectionData {
+    handle: Cell<Option<NetConnectionHandle>>,
+}
+
+#[derive(Copy, Clone, Debug, Collect)]
+#[collect(no_drop)]
+pub struct NetConnection<'gc>(Gc<'gc, NetConnectionData>);
+
+impl<'gc> NetConnection<'gc> {
+    pub fn handle(&self) -> Option<NetConnectionHandle> {
+        self.0.handle.get()
+    }
+
+    pub fn set_handle(&self, handle: Option<NetConnectionHandle>) -> Option<NetConnectionHandle> {
+        self.0.handle.replace(handle)
+    }
+
+    pub fn cast(value: Value<'gc>) -> Option<Self> {
+        if let Value::Object(object) = value {
+            if let NativeObject::NetConnection(net_connection) = object.native() {
+                return Some(net_connection);
+            }
+        }
+        None
+    }
+
+    pub fn on_status_event(
+        context: &mut UpdateContext<'_, 'gc>,
+        this: Object<'gc>,
+        code: &'static str,
+    ) -> Result<(), Error<'gc>> {
+        let Some(root_clip) = context.stage.root_clip() else {
+            tracing::warn!("Ignored NetConnection callback as there's no root movie");
+            return Ok(());
+        };
+        let mut activation = Activation::from_nothing(
+            context.reborrow(),
+            ActivationIdentifier::root("[NetConnection connect]"),
+            root_clip,
+        );
+        let constructor = activation.context.avm1.prototypes().object_constructor;
+        let event = constructor
+            .construct(&mut activation, &[])?
+            .coerce_to_object(&mut activation);
+        event.set("code", code.into(), &mut activation)?;
+        event.set("level", "status".into(), &mut activation)?;
+        this.call_method(
+            "onStatus".into(),
+            &[event.into()],
+            &mut activation,
+            ExecutionReason::Special,
+        )?;
+        Ok(())
+    }
+
+    // [NA] I have no idea why this is a thing. It's similar in AVM2 too.
```

Comment:
We confirmed that in AVM2 this is an event that sends the metadata script data. Have you checked that it's the same in AS2?

Reply:
I have no idea. How/where did you confirm that? I wasn't aware!

This is the code we have for avm2, which is as I originally wrote it still:
```actionscript
                    let event = Avm2EventObject::net_status_event(
                        &mut activation,
                        "netStatus",
                        vec![
                            ("code", ""),
                            ("description", ""),
                            ("details", ""),
                            ("level", "status"),
                        ],
                    );
```

For AVM1 it seems to just call the method with undefined.